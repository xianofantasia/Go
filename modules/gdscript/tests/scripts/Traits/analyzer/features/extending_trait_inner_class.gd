trait SomeTrait:
	class SomeInnerTrait:
		signal some_signal(param: Node)
		enum {
			enum_value
		}
		enum named_enum {
			named_enum_value
		}
		var some_var = "some var"
		func some_func():
			print("some func")

class SomeClass:
	uses SomeTrait
	class SomeInnerTrait:
		signal some_signal(param: Node2D)
		signal other_signal()
		enum {
			enum_value_2
		}
		enum named_enum {
			named_enum_value_2
		}
		var some_var = "overridden some var"
		var other_var = "some var in innerclass"
		func other_func():
			print("other func")

func test():
	var inner_class = SomeClass.SomeInnerTrait.new()
	inner_class.other_signal.connect(print.bind("some signal in innerclass"))
	inner_class.some_signal.emit()
	print(inner_class.enum_value)
	print(inner_class.enum_value_2)
	print(inner_class.named_enum.named_enum_value)
	print(inner_class.named_enum.named_enum_value_2)
	print(inner_class.some_var)
	print(inner_class.other_var)
	inner_class.some_func()
	inner_class.other_func()
	print("ok")
