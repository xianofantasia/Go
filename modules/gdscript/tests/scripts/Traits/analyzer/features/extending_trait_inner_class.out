GDTEST_OK
~~ WARNING at line 3: (UNUSED_SIGNAL) The signal "some_signal" is declared but never explicitly used in the class.
~~ WARNING at line 17: (UNUSED_SIGNAL) The signal "some_signal" is declared but never explicitly used in the class.
~~ WARNING at line 18: (UNUSED_SIGNAL) The signal "other_signal" is declared but never explicitly used in the class.
~~ WARNING at line 32: (UNSAFE_METHOD_ACCESS) The method "connect()" is not present on the inferred type "Variant" (but may be present on a subtype).
~~ WARNING at line 33: (UNSAFE_METHOD_ACCESS) The method "emit()" is not present on the inferred type "Variant" (but may be present on a subtype).
0
0
0
1
overridden some var
some var in innerclass
some func
other func
ok
